src/conf_mode/system_console.py

```diff
@@ -98,15 +97,29 @@ def generate(console):
     if not os.path.isfile(grub_config):
         return None
 
-    # stdin/stdout are redirected in replace_in_file(), thus print() is fine
+    with open(grub_config, 'r') as f:
+        lines = f.readlines()
+
     p = re.compile(r'^(.* console=ttyS0),[0-9]+(.*)$')
-    for line in replace_in_file(grub_config, inplace=True):
+    write = False
+    newlines = []
+    for line in lines:
         if line.startswith('serial --unit'):
-            line = f'serial --unit=0 --speed={speed}\n'
+            newline = f'serial --unit=0 --speed={speed}\n'
         elif p.match(line):
-            line = '{},{}{}\n'.format(p.search(line)[1], speed, p.search(line)[2])
+            newline = '{},{}{}\n'.format(p.search(line)[1], speed, p.search(line)[2])
+        else:
+            newline = line
+
+        if newline != line:
+            write = True
+
+        newlines.append(newline)
 
-        print(line, end='')
+    if write:
+        with open(grub_config, 'w') as f:
```

Comment:
As I understand it you do an in memory replacement instead of mangling the file on the filesystem line by line, correct?

If this is the case can we please also wrap the write with `vyos.util.write_file()`?

Reply:
Correct.

Done.